### ISSUE TITLE
upsert-on should optionally MERGE data based on timestamps

### ISSUE THREAD

**Issue by seamusabshere** [faradayio/dbcrossbar#58]
e.g. `upsert-on:id:updated_at`

then the MERGE would be

```
            r#"
MERGE INTO {dest_table} AS dest
USING {temp_table} AS temp
ON
    {key_comparisons}
WHEN MATCHED AND temp.updated_at > dest.updated_at THEN UPDATE SET
    {updates}
WHEN NOT MATCHED THEN INSERT (
    {columns}
) VALUES (
    {values}
)"#,
```

i added `AND temp.updated_at > dest.updated_at `

**Comment by emk** [comment]
Honestly, I feel like this may be starting to get a bit out of scope. `upsert-on:key1,key2` is a nice, clear behavior, but once you start adding in more complicated conditions, it might be better to write the data to a temporary table using `dbcrossbar`, and then do the upsert manually.

Also, if this is intended solely as an optimization, I'm not sure it buys much, because BigQuery will need to completely rewrite the whole table anyway (unless you have date partitioning working, but partitioning on `updated_at` would be weird).

TL;DR: Looks like a can of worms and I'm not entirely sure we want to open it.

**Comment by emk** [comment]
Or as another alternative, you could always use `--where "updated_at > $LAST_UPDATE_TIME"` when invoking `dbcrossbar`, to avoid copying records you know you already have. This would break if there are back-dated records, but if there are back-dated records, you probably want to do a full copy anyway.
